server/src/main/java/io/crate/execution/engine/indexing/GroupRowsByShard.java

```diff
@@ -68,7 +69,10 @@ public final class GroupRowsByShard<TReq extends ShardRequest<TReq, TItem>, TIte
     private final UnsafeArrayRow spareRow = new UnsafeArrayRow();
     private Object[] spareCells;
 
+    private final BiConsumer<String, IndexItem> constraintsChecker;
+
     public GroupRowsByShard(ClusterService clusterService,
+                            @Nullable BiConsumer<String, IndexItem> constraintsChecker,
```

Comment:
same here, I think passing in a dummy consumer instead of a null would avoid any mistake (null check)